.github/workflows/rust.yml

```diff
@@ -32,20 +31,55 @@ jobs:
           components: clippy, rustfmt
       - uses: Swatinem/rust-cache@v2
         with:
-          key: ${{ matrix.pwd }}
-          workspaces: ${{ matrix.pwd }}
+          key: .
+          workspaces: .
       - name: clippy
-        working-directory: ${{ matrix.pwd }}
+        working-directory: .
+        if: ${{ matrix.pwd == '.' }}
+        run: |
+          cargo clippy --all --all-targets --features sqlx-runtime-${{ matrix.runtime }}-${{ matrix.tls }},mssql,mysql,postgres,sqlite -- -Dwarnings
+      - name: clippy_examples
+        working-directory: .
+        if: ${{ matrix.pwd == 'examples' }}
         run: |
           cargo clippy --all --all-targets --all-features -- -Dwarnings
       - name: rustfmt
-        working-directory: ${{ matrix.pwd }}
+        working-directory: .
+        run: |
+          cargo fmt --all -- --check
+
+  check-examples:
```

Comment:
Is it necessary to break out examples this way?

Reply:
I believe so. The `examples` sub-crate doesn't have any features, so `--features sqlx-runtime-${{ matrix.runtime }}-${{ matrix.tls }},mssql,mysql,postgres,sqlite` raises an error. It worked before, because `--all-features` can mean "no features", but not anymore. I used a branch statement (and looking at it now, I see I've forgotten to remove it), but that still ran the same `exampes` test 6 times. I thought extracting it into its own job would be more efficient.